ext/semian/sysv_semaphores.c

```diff
@@ -130,24 +131,28 @@ get_semaphore(int key)
   return semget(key, SI_NUM_SEMAPHORES, 0);
 }
 
-void *
-acquire_semaphore_without_gvl(void *p)
+void
+acquire_semaphore_without_gvl(semian_operation_t *operation)
 {
-  WITHOUT_GVL(acquire_semaphore, p, RUBY_UBF_IO, NULL);
-  return NULL;
+  WITHOUT_GVL(acquire_semaphore, operation, RUBY_UBF_IO, NULL);
 }
 
 static void *
 acquire_semaphore(void *p)
```

Comment:
I assume this was originally a semian_resource_t? If so, then this should be a fair bit simpler if you just add the timing as a member to that struct.